### PR TITLE
[Booking_flow]Booking service repository

### PR DESCRIPTION
**Pull request by DevIgork**

### CODE REVIEW

File: src/main/java/com/application/bookingservice/service/booking/BookingServiceImpl.java [hunk 1]
```diff
@@ -11,36 +19,81 @@
...
     public BookingResponseDto findById(Long id) {
-        return null;
+        return bookingMapper.toDto(bookingRepository.findById(id)
+                .orElseThrow(
+                        () -> new EntityNotFoundException("Can't find booking by id : " + id))
+                );
     }
 
     @Override
     public BookingResponseDto updateById(Long id, BookingUpdateRequestDto bookingUpdateRequestDto) {
```
Comment (fedorovychh) [inline src/main/java/com/application/bookingservice/service/booking/BookingServiceImpl.java:54]: in such classes we can write just requestDto

**Review by fedorovychh — APPROVED** [review]
GJ. Add fetch please

File: src/main/java/com/application/bookingservice/service/booking/BookingServiceImpl.java [hunk 2]
```diff
@@ -11,36 +19,81 @@
...
+                bookingStatus,
+                pageable).stream()
+                .map(bookingMapper::toDto)
+                .toList();
+    }
+
+    @Override
+    public BookingResponseDto updateStatus(Long id, BookingUpdateStatusRequestDto requestDto) {
+        Booking booking = bookingRepository.findById(id).orElseThrow(
+                () -> new EntityNotFoundException("can't find booking by id :" + id)
```
Comment (nshtykh) [inline src/main/java/com/application/bookingservice/service/booking/BookingServiceImpl.java:94]: Please make constants for exception messages

File: src/main/java/com/application/bookingservice/service/booking/BookingServiceImpl.java [hunk 3]
```diff
@@ -11,36 +19,81 @@
...
-        return null;
+        Booking.Status bookingStatus = null;
+        for (Booking.Status currentStatus : Booking.Status.values()) {
+            if (currentStatus.name().equals(status)) {
+                bookingStatus = currentStatus;
+                break;
+            }
+        }
+        if (bookingStatus == null) {
+            throw new RuntimeException("not valid status : " + status);
```
Comment (nshtykh) [inline src/main/java/com/application/bookingservice/service/booking/BookingServiceImpl.java:81]: Please make constants for exception messages

File: src/main/java/com/application/bookingservice/service/booking/BookingServiceImpl.java [hunk 4]
```diff
@@ -11,36 +19,81 @@
...
+                .map(bookingMapper::toDto)
+                .toList();
     }
 
     @Override
     public BookingResponseDto findById(Long id) {
-        return null;
+        return bookingMapper.toDto(bookingRepository.findById(id)
+                .orElseThrow(
+                        () -> new EntityNotFoundException("Can't find booking by id : " + id))
```
Comment (nshtykh) [inline src/main/java/com/application/bookingservice/service/booking/BookingServiceImpl.java:49]: Please make constants for exception messages

**Review by nshtykh — APPROVED** [review]

**Pull request merged**